Add Github Actions CI script

oddhack opened the pull request:
To build HTML and PDF specs on push or manual selection.

Ready to merge - this has no impact on the actual spec.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/KhronosGroup/SYCL-Docs?pullRequest=117) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/KhronosGroup/SYCL-Docs?pullRequest=117) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/KhronosGroup/SYCL-Docs?pullRequest=117) it.</sub>

oddhack commented:
This is weird, Actions complains about not finding some of the intermediate image files and fails - but when I run exactly the same build on exactly the same branch using exactly the same Docker image locally, it works. It also works in internal gitlab CI using the same image. This may be painful to debug.

oddhack commented:
@keryell @gmlueck I put in a workaround to allow CI to work until there's a fix for the asciidoctor path problem.

The repo README is entirely incorrect as it refers to gitlab CI and a different Docker image (I have refactored and renamed the image to not imply Vulkan specific contents). If you want I can fix that up here before merging. We should probably have two sections, one for gitlab and one for github.

keryell commented:
> @keryell @gmlueck I put in a workaround to allow CI to work until there's a fix for the asciidoctor path problem.

It looks like it work!

> The repo README is entirely incorrect as it refers to gitlab CI and a different Docker image (I have refactored and renamed the image to not imply Vulkan specific contents). If you want I can fix that up here before merging. We should probably have two sections, one for gitlab and one for github.

Yes, probably having 2 sections is the way to go.
Thanks.

oddhack commented:
@keryell @gmlueck I've updated the README. This can be merged at any time.

oddhack commented:
@keryell note as well that the Docker image now being used has been updated to the current rouge gem, so should not need the 'dock' target.

gmlueck commented on .github/workflows/CI.yml:
```diff
@@ -0,0 +1,38 @@
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
```
Is the comment correct here that the workflow is triggered on both "push" and "pull"?  I don't know this yaml file scheme, but since it just says "push:" here, it makes me think the action is triggered only on "push" requests.

Also out of curiosity, why is the workflow limited to just the "master" branch?  I don't see that branch name anywhere.  Is it the default?

oddhack replied:
Good catch, thanks. Those are left over comments from a different branch I copied a starting point from.

oddhack replied:
Good catch, thanks. Those are left over comments from a different branch I copied a starting point from.

oddhack replied:
N.b. gitlab CI YAML is bad but github Actions is worse.

keryell reviewed (commented):
> @keryell note as well that the Docker image now being used has been updated to the current rouge gem, so should not need the 'dock' target.

Yes, I think it was not used on the CI, just for local developments.

oddhack commented:
> > @keryell note as well that the Docker image now being used has been updated to the current rouge gem, so should not need the 'dock' target.
> 
> Yes, I think it was not used on the CI, just for local developments.

The gem repository has updated to 3.26.0 - is something even newer needed?

keryell commented:
Thanks!
